ngraph/core/src/op/util/recurrent_sequence.cpp

```diff
@@ -0,0 +1,72 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "recurrent_sequence.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+void op::util::validate_seq_input_rank_dimension(const std::vector<ngraph::PartialShape>& input)
+{
+    enum
+    {
+        X,
+        initial_hidden_state,
+        sequence_lengths,
+        W,
+        R,
+        B
+    };
```

Comment:
Why do you need this local enumeration?
Looks like you use only part of values from this enum.

Reply:
https://github.com/openvinotoolkit/openvino/pull/1683